src/pgrn-custom-scan.c

```diff
@@ -147,9 +149,42 @@ PGrnCreateCustomScanState(CustomScan *cscan)
 	return (Node *) &(state->parent);
 }
 
+static Relation
+PGrnChooseIndex(Relation table)
+{
+	// todo: Support pgroonga_condition() index specification.
+	// todo: Implementation of the logic for choosing which index to use.
+	ListCell *cell;
+	List *indexes;
+
+	if (!table)
+		return NULL;
+
+	indexes = RelationGetIndexList(table);
+	foreach (cell, indexes)
+	{
+		Oid indexId = lfirst_oid(cell);
+
+		Relation index = RelationIdGetRelation(indexId);
+		if (!PGrnIndexIsPGroonga(index))
+		{
+			RelationClose(index);
+			continue;
+		}
+		return index;
+	}
+	return NULL;
+}
+
 static void
-PGrnBeginCustomScan(CustomScanState *cscanstate, EState *estate, int eflags)
+PGrnBeginCustomScan(CustomScanState *customScanState,
+					EState *estate,
+					int eflags)
 {
+	Relation index = PGrnChooseIndex(customScanState->ss.ss_currentRelation);
```

Comment:
```suggestion
	Relation index = PGrnChooseIndex(customScanState->ss.ss_currentRelation);
	if (!index)
	  return;
```

Probably, if `index` is `NULL`, PGroonga crash in `PGrnLookupSourcesTable()`